geopandas/geodataframe.py

```diff
@@ -1422,6 +1408,35 @@ def apply(self, func, axis=0, raw=False, result_type=None, args=(), **kwargs):
 
     @property
     def _constructor(self):
+        geometry_default = getattr(self, "_geometry_column_name")
+        crs_default = getattr(self, "crs")
+
+        def _geodataframe_constructor_with_fallback(
+            data=None, index=None, crs=crs_default, geometry=geometry_default, **kwargs
+        ):
+            df = pd.DataFrame(data, index=index)
```

Comment:
```suggestion
            df = pd.DataFrame(data, index=index, **kwargs)
```

(might not be used in practice by pandas, but let's pass it through to be sure)